server/src/test/java/com/hedera/block/server/persistence/storage/read/BlockAsLocalFileReaderTest.java

```diff
@@ -29,25 +51,64 @@
  * Tests for the {@link BlockAsLocalFileReader} class.
  */
 class BlockAsLocalFileReaderTest {
+    private BlockAsLocalFileWriter blockAsLocalFileWriterMock;
+    private BlockAsLocalFilePathResolver blockAsLocalFileResolverMock;
+    private BlockNodeContext blockNodeContext;
+    private PersistenceStorageConfig testConfig;
     private BlockAsLocalFileReader toTest;
 
+    @TempDir
+    private Path testLiveRootPath;
+
     @BeforeEach
-    void setUp() {
-        toTest = BlockAsLocalFileReader.newInstance();
+    void setUp() throws IOException {
+        blockNodeContext = TestConfigUtil.getTestBlockNodeContext(
+                Map.of(PERSISTENCE_STORAGE_LIVE_ROOT_PATH_KEY, testLiveRootPath.toString()));
+        testConfig = blockNodeContext.configuration().getConfigData(PersistenceStorageConfig.class);
+
+        final String testConfigLiveRootPath = testConfig.liveRootPath();
+        assertThat(testConfigLiveRootPath).isEqualTo(testLiveRootPath.toString());
+
+        blockAsLocalFileResolverMock = spy(BlockAsLocalFilePathResolver.of(testLiveRootPath));
+        blockAsLocalFileWriterMock = spy(BlockAsLocalFileWriter.of(blockNodeContext, blockAsLocalFileResolverMock));
+        toTest = BlockAsLocalFileReader.of(blockAsLocalFileResolverMock);
     }
 
     /**
      * This test aims to verify that the
      * {@link BlockAsLocalFileReader#read(long)} correctly reads a block with
      * a given block number.
-     *
-     * @param toRead parameterized, block number
      */
-    @ParameterizedTest
-    @MethodSource("validBlockNumbers")
-    void testSuccessfulBlockRead(final long toRead) {
-        // todo currently throws UnsupportedOperationException
-        assertThatExceptionOfType(UnsupportedOperationException.class).isThrownBy(() -> toTest.read(toRead));
+    @Test
+    void testSuccessfulBlockRead() throws IOException, ParseException {
+        final List<BlockItemUnparsed> blockItemUnparsed = PersistTestUtils.generateBlockItemsUnparsed(1);
+        final Optional<List<BlockItemUnparsed>> written = blockAsLocalFileWriterMock.write(blockItemUnparsed);
+
+        assertThat(written).isNotNull().isPresent();
+
+        final Optional<BlockUnparsed> actual = toTest.read(1);
+        assertThat(actual)
+                .isNotNull()
+                .isPresent()
+                .get(InstanceOfAssertFactories.type(BlockUnparsed.class))
+                .isNotNull()
+                .isExactlyInstanceOf(BlockUnparsed.class)
+                .returns(1L, from(blockUnparsed -> {
+                    try {
+                        return BlockHeader.PROTOBUF
+                                .parse(Objects.requireNonNull(
+                                        blockUnparsed.blockItems().getFirst().blockHeader()))
+                                .number();
+                    } catch (final ParseException e) {
+                        throw new RuntimeException(e);
+                    }
```

Comment:
FYI, unit tests generally should not catch exceptions; let the exception be thrown from the test method and the test runner will give you better information in the test report.

Reply:
I agree, however the `returns` test requires a `Function` which unfortunately does not throw, and the `parse` method throws an unchecked exception. I do re-throw it immediately however so we do not silence the exception (as it should be the case).